modules/wri_event_callout/src/Plugin/Field/FieldFormatter/EventCalloutBehaviorFormatter.php

```diff
@@ -0,0 +1,263 @@
+
+    return $elements;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function settingsSummary(): array {
+    $summary = parent::settingsSummary();
+
+    $summary[] = $this->t('View: @view (@display)', [
+      '@view' => $this->getSetting('view_id'),
+      '@display' => $this->getSetting('display_id'),
+    ]);
+
+    $summary[] = $this->t('Event date field: @field', [
+      '@field' => $this->getSetting('event_date_field'),
+    ]);
+
+    return $summary;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(FieldItemListInterface $items, $langcode): array {
+    $elements = [];
+
+    $entity = $items->getEntity();
+    if (!$entity instanceof BlockContentInterface || $entity->bundle() !== 'upcoming_event') {
+      return $elements;
+    }
+
+    // Behavior value:
+    // - show_next_event
+    // - link_to_recording
+    // - hide_block
+    // Defaults to show_next_event (matches field default).
+    $behavior = $items->isEmpty() ? 'show_next_event' : (string) $items->first()->value;
+
+    /** @var \Drupal\node\NodeInterface|null $event */
+    $event = NULL;
+    if ($entity->hasField('field_event_to_feature') && !$entity->get('field_event_to_feature')->isEmpty()) {
+      $event = $entity->get('field_event_to_feature')->entity;
+    }
+
+    // Case 1: no explicit event chosen, always use "next upcoming by topic".
+    if (!$event instanceof NodeInterface) {
+      $build = $this->buildUpcomingByTopic($entity);
+      if (!empty($build)) {
+        $elements[0] = $build;
+      }
+      return $elements;
+    }
+
+    // Case 2: explicit event exists and is in the future, always show it.
+    if ($this->isEventUpcoming($event)) {
+      $elements[0] = $this->buildEventCallout($event);
+      return $elements;
+    }
+
+    // From here on, the event is over.
+    switch ($behavior) {
+      case 'link_to_recording':
+        // Show the chosen (past) event in Callout view mode.
+        $elements[0] = $this->buildEventCallout($event);
+        return $elements;
+
+      case 'show_next_event':
+        // Show the next upcoming event in the chosen topic.
+        $build = $this->buildUpcomingByTopic($entity);
+        if (!empty($build)) {
+          $elements[0] = $build;
+        }
+        return $elements;
+
+      case 'hide_block':
+      default:
+        // Hide block, or fallback when everything else fails.
+        return $elements;
+    }
+  }
+
+  /**
+   * Check if an event is considered "upcoming".
+   */
+  protected function isEventUpcoming(NodeInterface $event): bool {
+    $date_field = $this->getSetting('event_date_field');
+
+    if (!$event->hasField($date_field) || $event->get($date_field)->isEmpty()) {
+      return FALSE;
+    }
+
+    // For datetime or daterange, the first item's "value" is usually the start.
+    $item = $event->get($date_field)->first();
+    $value = $item->value ?? NULL;
+    if (!$value) {
+      return FALSE;
+    }
+
+    $timestamp = strtotime($value);
```

Comment:
For me, `$value` is already a timestamp, like `1289361600` so the next line is always true. This means that if I choose an event that's still upcoming AND a topic, it falls back to the topic view, even if my event is upcoming.